SSA/Core/ErasedContext.lean

```diff
@@ -250,6 +250,16 @@ theorem Valuation.snoc_toSnoc {Γ : Ctxt Ty} {t t' : Ty} (s : Γ.Valuation) (x :
     (v : Γ.Var t') : (s.snoc x) v.toSnoc = s v := by
   simp [Ctxt.Valuation.snoc]
 
+/-!
+# Helper to simplify context manipulation with toSnoc and variable access.
+-/
+/--  (ctx.snoc v₁) ⟨n+1, _⟩ = ctx ⟨n, _⟩ -/
+@[simp]
+theorem Valuation.snoc_eval {ty : Ty} (Γ : Ctxt Ty) (V : Γ.Valuation) (v : ⟦ty⟧)
+    (hvar : Ctxt.get? (Ctxt.snoc Γ ty) (n+1) = some var_val) :
+    (V.snoc v) ⟨n+1, hvar⟩ = V ⟨n, by simp [Ctxt.get?,Ctxt.snoc] at hvar; exact hvar⟩ :=
+  rfl
```

Comment:
I think this should be called `Valuation.snoc_succ` or something similar.

Somewhat related: we should pick a normal form between `V (Var.last ..)`/`V (Var.toSnoc ..)` and `V ⟨_, _⟩`. I would prefer the former, making this lemma unnecessary (with `snoc_toSnoc` already existing)

Reply:
Renamed to `snoc_succ`. I am not sure how to setup the `simp` normal form correctly, I tried adding more lemmas, but was unable to get it to normalize as I wanted to. Is there a tool to debug simp normal forms?

Reply:
None that I know of. Usually, when a `simp` lemma I expect to apply isn't working, I'll try using it explicitly with `rw` and try to debug from there

Reply:
We can defer this to an issue, I'm merging for now